Upstream-Status: Backport
Signed-off-by: Ross Burton <ross.burton@arm.com>

From 45fe65d934df2c938701f1c3cd3e0a6f3bdf25e6 Mon Sep 17 00:00:00 2001
From: Ross Burton <ross.burton@arm.com>
Date: Tue, 29 Jun 2021 18:10:40 +0100
Subject: [PATCH] tests/blkdiscard: check correct log file for errors

When checking if the discard ioctl is supported, look in stderr not stdout.

Signed-off-by: Ross Burton <ross.burton@arm.com>
---
 tests/ts/blkdiscard/offsets | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/ts/blkdiscard/offsets b/tests/ts/blkdiscard/offsets
index 9dddc35e2..5e724d51b 100755
--- a/tests/ts/blkdiscard/offsets
+++ b/tests/ts/blkdiscard/offsets
@@ -48,7 +48,7 @@ ts_log "testing offsets with full block size"
 run_tscmd $TS_CMD_BLKDISCARD -v $DEVICE
 if [ "$?" != "0" ]; then
 	# Skip the rest? For example loop backing files on NFS seem unsupported.
-	grep -q "BLKDISCARD ioctl failed: Operation not supported" "$TS_OUTPUT" \
+	grep -q "BLKDISCARD ioctl failed: Operation not supported" "$TS_ERRLOG" \
 		&& ts_skip "BLKDISCARD not supported"
 fi
 run_tscmd $TS_CMD_BLKDISCARD -v -o 1 $DEVICE
-- 
2.25.1

